libs/contrib/Data/List/Equalities.idr

```diff
@@ -115,3 +117,33 @@ export
 appendNonEmptyRightNotEq : (zs, xs : List a) -> NonEmpty xs => Not (zs = zs ++ xs)
 appendNonEmptyRightNotEq []      (_::_)  Refl impossible
 appendNonEmptyRightNotEq (_::zs) (x::xs) prf = appendNonEmptyRightNotEq zs (x::xs) $ snd $ biinjective prf
+
+export
+bindOnEmptyList : (f : a -> List b) -> ([] >>= f) === []
+bindOnEmptyList f = Refl
```

Comment:
Is this necessary, or merely for completeness' sake?
The equation holds judgementally, so it's probably easier to just use `Refl` instead of calling `bindOnEmptyList`.